Database.py

```diff
@@ -0,0 +1,33 @@
+from models import db, User
+from flask_sqlalchemy import SQLAlchemy
+from flask_bcrypt import Bcrypt
+
+bcrypt = Bcrypt()
+
+
+def connect_db(app):
+    """Connect to database."""
+
+    db.app = app
+    db.init_app(app)
+
+
+def hash_password(password):
+    return bcrypt.generate_password_hash(password).decode('UTF-8')
+
+
+def signup(username, password, first_name, last_name):
```

Comment:
This function is doing three things -- generating a password, creating a new user object, and adding it to the database. You could probably argue that creating the user object and adding it to the database belong together. Personally, I like those divided but with a framework like flask its pretty typical so in this context this is how I would write it like:
```py
def add_user(username, hashed_password, first_name, last_name):
    user = User(
        username=username,
        password=hashed_password,
        first_name=first_name,
        last_name=last_name)
    db.session.add(user)
    return user
```
Now, at most, its doing two things (which like we said is acceptable in flask in this context). This means you'll need to move the password hashing up to a higher layer in your api.